pkg/operator/encryption/deployer/unionrevisionedpod.go

```diff
@@ -0,0 +1,79 @@
+package deployer
+
+import (
+	"reflect"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/openshift/library-go/pkg/operator/encryption/encryptionconfig"
+	"github.com/openshift/library-go/pkg/operator/encryption/statemachine"
+)
+
+// UnionRevisionLabelPodDeployer provides unified state from multiple distinct deployers.
+type UnionRevisionLabelPodDeployer struct {
+	delegates []statemachine.Deployer
+}
+
+var _ statemachine.Deployer = &RevisionLabelPodDeployer{}
+
+// NewUnionRevisionLabelPodDeployer creates a deployer that returns a unified state from multiple distinct deployers.
+// That means:
+//  - none has reported an error
+//  - all have converged
+//  - all have observed exactly the same encryption configuration
+func NewUnionRevisionLabelPodDeployer(delegates ...statemachine.Deployer) *UnionRevisionLabelPodDeployer {
+	return &UnionRevisionLabelPodDeployer{delegates: delegates}
+}
+
+// DeployedEncryptionConfigSecret returns the actual encryption configuration across multiple deployers
+func (d *UnionRevisionLabelPodDeployer) DeployedEncryptionConfigSecret() (secret *corev1.Secret, converged bool, err error) {
```

Comment:
any reason why to use named return values?

Reply:
no reason indeed will remove, thx